astropy/visualization/wcsaxes/tests/test_coordinate_helpers.py

```diff
@@ -33,7 +30,8 @@ def test_getaxislabel(ignore_matplotlibrc):
 
 @pytest.fixture
 def ax():
-    fig = plt.figure()
+    fig = Figure()
+    _canvas = FigureCanvasAgg(fig)
```

Comment:
I don't grok setting `_canvas` and then not use it. Is it doing some mpl magic?

Reply:
well, technically, `plt.figure()` is the magic-doing one; it's really setting up the canvas for you (which a figure *requires* before it can be drawn to), but `Figure()` doesn't. `pyplot` is the high level interface, and replacing it with lower level API calls exposes the dirty details.